homeassistant/helpers/entity_platform.py

```diff
@@ -641,7 +641,19 @@ async def async_add_entities(
     @callback
     def _async_handle_interval_callback(self, now: datetime) -> None:
         """Update all the entity states in a single platform."""
-        self.hass.async_create_task(self._update_entity_states(now), eager_start=True)
+        if self.config_entry:
+            self.config_entry.async_create_periodic_task(
+                self.hass,
+                self._update_entity_states(now),
+                name=f"EntityPlatform poll {self.domain}.{self.platform_name}",
+                eager_start=True,
+            )
```

Comment:
Now that the polling gets cancelled when the config entry unloads, this reduces at least one place we will hit the guard that was introduced in #67750